src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs

```diff
@@ -91,9 +90,14 @@ public async Task StopAsync(CancellationToken cancellationToken = default)
                 // Fire IHostApplicationLifetime.Stopped
                 _applicationLifetime.NotifyStopped();
 
-                token.ThrowIfCancellationRequested();
-                await _hostLifetime.StopAsync(token).ConfigureAwait(false);
-
+                try
+                {
+                    await _hostLifetime.StopAsync(token).ConfigureAwait(false);
+                }
+                catch (Exception ex)
```

Comment:
Should we catch as `Exception` or a more specific type?

cc: @eerhardt @BrennanConroy

Reply:
`Exception` seems fine, hostlifetime can be anything and this is the same as what we do above with hostedservice.